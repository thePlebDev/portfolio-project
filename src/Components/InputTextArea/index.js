import React from 'react';
import styled from 'styled-components';

const TextArea = styled.textarea`
  width:70%;
  height:70px;
  margin-top:2rem;
  border:${props=>props.error ?'1px solid red':''};
  padding:5px;
  font-size:20px;
  @media only screen and (min-width: 1200px){
    width:53%;
  };
`


const InputTextArea =({name="enter name prop",error='enter error prop',value='enter value prop',onChange})=>{
  return(
    <div>
      <label for={name} data-testid="label"></label>
      <TextArea name={name} data-testid="textarea" id={name} error={error} value={value} onChange={(e)=>onChange(e)}placeholder="MESSAGE"></TextArea>
    </div>
  )
}


export default InputTextArea
