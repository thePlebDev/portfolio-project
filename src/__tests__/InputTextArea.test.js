import React from 'react';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import InputTextArea from '../Components/InputTextArea'

//1)snapshots
//2)renders
//3)props
//4)actions


describe('testing the InputTextArea component',()=>{
  let wrapper;
  beforeEach(()=>{
    wrapper = shallow(<InputTextArea/>)
  })
  it('should render the snapshot',()=>{
    const tree = renderer.create(<InputTextArea />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should render correctly',()=>{
    const input = wrapper.find('[data-testid="label"]')
    const textArea = wrapper.find('[data-testid="textarea"]')
    expect(input.length).toBe(1)
    expect(textArea.length).toBe(1)
  })
  it('should render the proper defualt props',()=>{
    const defaultName ="enter name prop"
    const defaultValue ='enter value prop'
    const defaultError = 'enter error prop'
    const input = wrapper.find('[data-testid="label"]')
    const textArea = wrapper.find('[data-testid="textarea"]')

    expect(input.props().for).toBe(defaultName)

    expect(textArea.props().name).toBe(defaultName)
    expect(textArea.props().value).toBe(defaultValue)
    expect(textArea.props().error).toBe(defaultError)
    expect(typeof textArea.props().onChange).toBe('function')

  })
  describe('testing the custom prop component',()=>{
    let wrapper;
    let mockOnChange = jest.fn(()=>{})
    beforeEach(()=>{
      wrapper = shallow(<InputTextArea name={'it'} value={'go'} error={'down'} onChange={mockOnChange}/>)
    })
    it('should be a component',()=>{
      const label = wrapper.find('[data-testid="label"]')
      const textarea = wrapper.find('[data-testid="textarea"]')

      expect(textarea.props().name).toBe('it')
      expect(textarea.props().value).toBe('go')
      expect(textarea.props().error).toBe('down')

      expect(label.props().for).toBe('it')

      textarea.simulate('change',{target:{value:'it do be like that sometimes'}})
      expect(mockOnChange.mock.calls.length).toBe(1)
    })
  })
})
