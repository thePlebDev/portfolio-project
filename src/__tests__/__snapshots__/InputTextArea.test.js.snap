// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the InputTextArea component should render the snapshot 1`] = `
<div>
  <label
    data-testid="label"
    for="enter name prop"
  />
  <textarea
    className="sc-bdnylx jMJXPG"
    data-testid="textarea"
    id="enter name prop"
    name="enter name prop"
    onChange={[Function]}
    placeholder="MESSAGE"
    value="enter value prop"
  />
</div>
`;
